File: LeetCode/LeetCode 226. 翻转二叉树 (dfs).cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* ans, TreeNode* root) {
        if(!root) return ;
        if(root->left) {
            ans->right = new TreeNode(root->left->val);
            dfs(ans->right, root->left);
        }
        if(root->right) {
            ans->left = new TreeNode(root->right->val);
            dfs(ans->left, root->right);
        }
        return ;
    }

    TreeNode* invertTree(TreeNode* root) {
        if(!root) return nullptr;
        TreeNode* ans = new TreeNode(root->val);
        dfs(ans, root);
        return ans;
    }
};

File: LeetCode/LeetCode 226. 翻转二叉树 (dfs).py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return None
        ans = TreeNode(root.val)

        def dfs(ans: TreeNode, root: TreeNode):
            if not root:
                return 
            if root.left:
                ans.right = TreeNode(root.left.val)
                dfs(ans.right, root.left)
            if root.right:
                ans.left = TreeNode(root.right.val)
                dfs(ans.left, root.right)
            return 
        
        dfs(ans, root)
        return ans